Bio/__init__.py

```diff
@@ -111,3 +114,11 @@ class BiopythonExperimentalWarning(BiopythonWarning):
     """
 
     pass
+
+
+_parent_dir = os.path.dirname(os.path.dirname(__file__))
+if os.path.exists(os.path.join(_parent_dir, "setup.py")):
```

Comment:
How about making this a little narrower, with checks for ``Bio/`` and ``BioSQL`` subdirectories? That ought to eliminate most false positives.

Reply:
Thanks Peter, can you give an example of the kind of false positives you are thinking of? This will help me understand how to change the code

Reply:
I think what Peter means is that there is a certain chance, that you are outside the Biopython source tree and still find a file named `setup.py` in your current folder, because this filename is not so rare. E.g. in my Python environment I have several (50+?) `setup.py` files. So testing for Biopython specific subfolders will ensure that you are in the Biopython source tree (or at least in something that looks very similar...).

Reply:
Naive question, is it possible that `Bio.__file__` does not contain `Bio/`?

@MarkusPiotrowski you have `setup.py` in `/path/lib/python3.X/site-packages/` or similar?

Reply:
> @MarkusPiotrowski you have setup.py in /path/lib/python3.X/site-packages/ or similar?

Yes, several, e.g. in `numpy` and `win32`.

Reply:
@MarkusPiotrowski So in `/site-packages/` you would have `setup.py`, `Bio/`, and `BioSQL/`?- so how would you distinguish this from the source tree?

To clarify, the `site-packages/` directory is `_parent_dir` for me when I'm *outside* the source tree. And i'm only checking for `_parent_dir/setup.py`, this is not like `_parent_dir/*/setup.py` (no subdirectories are checked)

What am i missing? I tried to follow https://github.com/biopython/biopython/issues/1991#issuecomment-482766471

Reply:
Maybe _I'm_ missing something or haven't thought it through . OK, if have no file `setup.py` directly in my ` /site-packages/` folder. But there are so many around, can you guarantee that there will be never one? 
But you are right, in this case `Bio/` and `BioSQL/` would also be present. What about a combination of `Bio`, `BioSQL`, `Tests`, `Doc` and `Scripts`? Or check in addition something like `if __file__.startswith("Bio\\__init__.py")` ?

Reply:
Perhaps you could look at the contents of setup.py to determine if it is the Biopython setup.py script. Or check for the presence and contents of the `PKG-INFO` file, which may be easier to parse.

Reply:
When in the source tree, `__file__` is still the full path for me it seems.

How about after checking for `setup.py` existence doing: `subprocess.check_output("python setup.py --name", shell=True) == b'biopython\n'` ?

Reply:
Yes, the conditional for this new warning will be triggered every single time someone starts up Python and import Biopython - so is should be as efficient as possible. On our current cluster file system I'm sometimes painfully aware of even the file access overhead. And I would definitely want to avoid the overhead of a subprocess call which I think would be several times slower.

Any check on the ``__file__`` variable will be trivial compared to checking for a file or folder on disk - so if we can put that first in any composite condition, that's worthwhile 👍 

Personally I would prioritise speed here, while trying to avoid false positive warnings.

Reply:
Thanks Peter, agreed that avoiding checking for files/directories would be better, but how can we check `__file__` variable? I think @mdehoon 's original suggestion to check for relative vs. absolute path is only for Python 2?

Reply:
There's no reaction emoji for a sad face. Confirming on macOS,

```
$ cd ~ && python -c "import Bio; print(Bio.__file__)"
/Library/Python/2.7/site-packages/Bio/__init__.pyc
$ cd ~/repositories/biopython && python -c "import Bio; print(Bio.__file__)"
Bio/__init__.py
```

vs:

```
$ cd ~ && python3 -c "import Bio; print(Bio.__file__)"
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/Bio/__init__.py
$ cd ~/repositories/biopython && python3 -c "import Bio; print(Bio.__file__)"
/Users/XXX/repositories/biopython/Bio/__init__.py
```

I guess checking for ``../setup.py`` is a reasonable compromise?